From 525c51997a36cdae4e024dc12be18814d7e4f87f Mon Sep 17 00:00:00 2001
From: Matteo Bertozzi <matteo.bertozzi@cloudera.com>
Date: Thu, 26 Sep 2013 15:21:02 +0100
Subject: [PATCH 02/86] CLOUDERA-BUILD add safety check to build to ensure JDK7 is being used

The default target is Java 6 but you must use JDK7 to compile
---
 pom.xml |   47 ++++++++++++++++++++++++++++++++++++++++++++---
 1 files changed, 44 insertions(+), 3 deletions(-)

diff --git a/pom.xml b/pom.xml
index 5b74245..81f8b3b 100644
--- a/pom.xml
+++ b/pom.xml
@@ -469,8 +469,8 @@
           <artifactId>maven-compiler-plugin</artifactId>
           <version>2.5.1</version>
           <configuration>
-            <source>${compileSource}</source>
-            <target>${compileSource}</target>
+            <source>${sourceJavaVersion}</source>
+            <target>${targetJavaVersion}</target>
             <showWarnings>true</showWarnings>
             <showDeprecation>false</showDeprecation>
             <compilerArgument>-Xlint:-options</compilerArgument>
@@ -862,6 +862,45 @@
       </plugin>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <version>1.0</version>
+        <inherited>false</inherited>
+        <configuration>
+          <rules>
+            <!--<requireMavenVersion>-->
+              <!--<version>[3.0.0,)</version>-->
+            <!--</requireMavenVersion>-->
+            <requireJavaVersion>
+              <version>[${javaVersion}.0,${javaVersion}.1000}]</version>
+            </requireJavaVersion>
+          </rules>
+        </configuration>
+        <executions>
+          <execution>
+            <id>clean</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <phase>pre-clean</phase>
+          </execution>
+          <execution>
+            <id>default</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <phase>validate</phase>
+          </execution>
+          <execution>
+            <id>site</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <phase>pre-site</phase>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-site-plugin</artifactId>
         <version>${maven.site.version}</version>
         <inherited>false</inherited>
@@ -888,8 +927,10 @@
       yyyy-MM-dd'T'HH:mm
     </maven.build.timestamp.format>
     <buildDate>${maven.build.timestamp}</buildDate>
-    <compileSource>1.6</compileSource>
     <!-- Dependencies -->
+    <javaVersion>1.7</javaVersion>
+    <targetJavaVersion>1.6</targetJavaVersion>
+    <sourceJavaVersion>${targetJavaVersion}</sourceJavaVersion>
     <hadoop-two.version>${cdh.hadoop.version}</hadoop-two.version>
     <hadoop-one.version>1.1.2</hadoop-one.version>
     <commons-cli.version>1.2</commons-cli.version>
-- 
1.7.0.4

